pkg/helm/option.go

```diff
@@ -361,6 +368,80 @@ func WithMaxHistory(max int32) HistoryOption {
 
 // NewContext creates a versioned context.
 func NewContext() context.Context {
-	md := metadata.Pairs("x-helm-api-client", version.Version)
+	//md := metadata.Pairs("x-helm-api-client", version.Version,)
+	contextData := getMapFromKubeConfig()
+	contextData = append(contextData, "x-helm-api-client")
+	contextData = append(contextData, version.Version)
+	md := metadata.Pairs(contextData...)
 	return metadata.NewContext(context.TODO(), md)
 }
+
+func getMapFromKubeConfig() []string {
+	var contextData []string
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	flags.SetNormalizeFunc(utilflag.WarnWordSepNormalizeFunc) // Warn for "_" flags
+	clientConfig := cmdutil.DefaultClientConfig(flags)
```

Comment:
We need to support all possible auth types: https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/api/v1/types.go#L84

- Read from File.
- What happens when auth-provider: gcp is used (GKE).

The best thing will be kubectl lib to process KubeConfig file, and then we set headers from that. Then we stay future proof.
cc: @sadlil